container/container_unix.go

```diff
@@ -220,7 +221,9 @@ func (container *Container) UnmountIpcMounts(unmount func(pth string) error) {
 			warnings = append(warnings, err.Error())
 		} else if shmPath != "" {
 			if err := unmount(shmPath); err != nil && !os.IsNotExist(err) {
-				warnings = append(warnings, fmt.Sprintf("failed to umount %s: %v", shmPath, err))
+				if mounted, mErr := mount.Mounted(shmPath); mounted || mErr != nil {
```

Comment:
@cpuguy83 looks like this is now doing the same as [`mount.Unmount()`](https://github.com/moby/moby/blob/54dcbab25ea4771da303fa95e0c26f2d39487b49/pkg/mount/mount.go#L53-L60), only in reverse order (first try `syscall.Unmount()`, then check if it was mounted - I assume that's a performance optimisation) - should we move that optimisation in `mount.Unmount()` and use it here?

It looks like the `unmount` argument for `UnmountIpcMounts()` is not really needed anymore (it looks like it was was added in https://github.com/moby/moby/commit/78bd17e805b7514505455b10f2fd90962505a3ff to allow different unmount methods in different parts of the code)